src/software/new_geom/BUILD

```diff
@@ -22,6 +22,52 @@ cc_library(
         "shape.h",
         "vector.h",
     ],
+    deps = [
+        ":closest_point",
+        ":distance",
+        "@boost//:geometry",
+        "@boost//:polygon",
+    ],
+)
+
+cc_library(
+    name = "closest_point",
+    srcs = [
+        "line.cpp",
+        "point.cpp",
```

Comment:
If there are multiple libraries that depend on `line`, `point`, etc., then we should create libraries for `line`, `point`, etc. and then use `deps` to depend on them. This way we avoid unnecesarry re-compiling of files and make it way easier to manage changing dependencies in the future (consider the case where `line` depends on some new library).

Reply:
Updated BUILD!

Reply:
Looking good, left one more comment below.